### PR TITLE
Further OSPEX updates

### PR DESCRIPTION
**Pull request by grazwegian**
Updates mainly dealing with pixel and detector selection and labelling.

### CODE REVIEW

File: stix/idl/processing/spectrogram/stx_fsw_sd_spectrogram2ospex.pro [hunk 1]
```diff
@@ -125,15 +128,31 @@ function stx_fsw_sd_spectrogram2ospex, spectrogram, specpar = specpar, time_shif
...
+      ;xspec in gneral works with energy depandent systematic errors
+      e_axis = spectrum_in.e_axis
+      n_energies = n_elements(e_axis.mean)
+      sys_err  = fltarr(n_energies)
+
+      idx_below10kev = where(e_axis.mean lt 10, cb10)
+      sys_err[*] = 0.03
+      if cb10 gt 0 then sys_err[idx_below10kev] = 0.05
+      idx_below7kev = where(e_axis.mean lt 7, cb7)
+      if cb7 gt 0 then sys_err[idx_below7kev] = 0.07
```
Comment (samaloney) [inline stix/idl/processing/spectrogram/stx_fsw_sd_spectrogram2ospex.pro:143]: So we have different systematic error if different energy ranges?

**Review by samaloney — APPROVED** [review]
Looks good, the pixel and detector mask stuff is nice.

File: stix/idl/processing/spectrogram/stx_fsw_sd_spectrogram2ospex.pro (hunk 1 above)
Comment (grazwegian) [reply stix/idl/processing/spectrogram/stx_fsw_sd_spectrogram2ospex.pro:143]: Yes, this is currently only relevant for XSPEC as OSPEX can only handle a single value but the idea is to capture our confidence in the low energy response compared with the higher energies.

**Review by ennosigaeus — APPROVED** [review]
LGTM

**Pull request merged**